tools/deploy.sh

```diff
@@ -90,30 +90,50 @@ RESTART_CONTAINER_CERTIFICATE_UPDATED=${RESTART_CONTAINER_CERTIFICATE_UPDATED:-"
 export NAMEPREFIX=${NAMEPREFIX:-"baremetal-operator"}
 
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+IRONIC_BASIC_AUTH_COMPONENT="${SCRIPTDIR}/ironic-deployment/components/basic-auth"
+TEMP_IRONIC_OVERLAY="${SCRIPTDIR}/ironic-deployment/overlays/temp"
+rm -rf "${TEMP_IRONIC_OVERLAY}"
+mkdir -p "${TEMP_IRONIC_OVERLAY}"
 
-KUSTOMIZE="tools/bin/kustomize"
+KUSTOMIZE="${SCRIPTDIR}/tools/bin/kustomize"
 make -C "$(dirname "$0")/.." "${KUSTOMIZE}"
 
+# Create a temporary overlay where we can make changes.
+pushd "${TEMP_IRONIC_OVERLAY}"
+${KUSTOMIZE} create --resources=../../../config/namespace \
+  --namespace=baremetal-operator-system --nameprefix=baremetal-operator-
+${KUSTOMIZE} edit add secret mariadb-password --from-literal=password=changeme
+
+# Basic-auth + TLS is special since TLS also means reverse proxy, which affects basic-auth.
+# Therefore we have an overlay that we use as base for this case.
+if [ "${DEPLOY_BASIC_AUTH}" == "true" ] && [ "${DEPLOY_TLS}" == "true" ]; then
+    ${KUSTOMIZE} edit add resource ../../overlays/basic-auth_tls
+else
+    ${KUSTOMIZE} edit add resource ../../base
+    if [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
+        ${KUSTOMIZE} edit add component ../../components/basic-auth
+    elif [ "${DEPLOY_TLS}" == "true" ]; then
+        ${KUSTOMIZE} edit add component ../../components/tls
+    fi
+fi
+
+if [ "${DEPLOY_KEEPALIVED}" == "true" ]; then
+    ${KUSTOMIZE} edit add component ../../components/keepalived
+fi
+
 if [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
     BMO_SCENARIO="${SCRIPTDIR}/config/basic-auth"
```

Comment:
Good point! I'm reworking this a bit now, hopefully it will be less duplication of conditions. At first my idea was to keep ironic and BMO separate to make it clearer, but I think it didn't really work well.

Reply:
Done now, PTAL!

Reply:
Thanks, it looks good to me 👍🏼